app/controller/button/DigitizeButtonController.js

```diff
@@ -243,21 +283,163 @@ Ext.define('CpsiMapview.controller.button.DigitizeButtonController', {
     },
 
     /**
-     * Handles the draw end event by getting the features and passing them
-     * to the `prepareRequestParams` function
-     * @param {DrawEvent} evt The OpenLayers draw event containing the features
+     * returns all features in the active group
+     * @returns {ol.Feature[]}
+     */
+    getActiveGroupFeatures: function () {
+        var me = this;
+        return this.resultLayer.getSource().getFeatures()
+            .filter(function (feature) {
+                return feature.get('group') === me.activeGroupIdx;
+            });
+    },
+
+    getSolverPoints: function () {
+        return this.getActiveGroupFeatures()
+            .filter(function (feature) {
+                return feature.getGeometry() instanceof ol.geom.Point;
+            })
+            .sort(function (a, b) {
+                return a.get('index') - b.get('index');
+            });
+    },
+
+    /**
+     * Handles the drawend event and gets the netsolver result which is passed to `handleFinalResult`
+     * @param {ol.interaction.Draw.Event} evt The OpenLayers draw event containing the features
      */
     handleDrawEnd: function (evt) {
-        // evt.feature if filled after drawend, only contains current finished feature.
-        // evt.features is set on modifyend and will contain all the current features of that geom-type
-        var feat = evt.feature ? evt.feature : evt.features.getArray();
-        this.prepareRequestParams(feat);
+        var me = this;
+        var view = me.getView();
+
+        var resultPromise;
+
+        switch (view.getType()) {
+            case 'Point':
+                var points = me.getSolverPoints();
+                resultPromise = me.getNetByPoints(points.concat([evt.feature]));
+                break;
+            case 'Polygon':
+                resultPromise = me.getNetByPolygon(evt.feature);
+                break;
+            case 'Circle':
+                resultPromise = me.getNetByCircle(evt.feature);
+                break;
+            default:
+                BasiGX.warn('Please implement your custom handler here for ' + view.getType());
+                return;
+        }
+
+        resultPromise
+            .then(me.handleFinalResult.bind(me))
+            .then(undefined, function (err) {
+                Ext.log.error(err);
+            })
+            .then(me.updateDrawSource.bind(me));
+    },
+
+
+    /**
+     * Handles the modifyend event and gets the netsolver result which is passed to `handleFinalResult`
+     * @param {ol.interaction.Modify.Event} evt The OpenLayers modify event containing the features
+     */
+    handleModifyEnd: function (evt) {
+        var me = this;
+        var view = me.getView();
+
+        // find modified feature
+        var drawFeature = me.map.getFeaturesAtPixel(evt.mapBrowserEvent.pixel, {
+            layerFilter: function (layer) {
+                return layer === me.drawLayer;
+            }
+        })[0];
```

Comment:
`drawFeature` could be null (happened during testing)

Reply:
did this happen during deletion of a polygon vertex or any other modification? If it happened during any other modification I have to investigate further.